hudi-common/src/main/java/org/apache/hudi/common/engine/HoodieReaderContext.java

```diff
@@ -165,10 +167,50 @@ public void setShouldMergeUseRecordPosition(boolean shouldMergeUseRecordPosition
    * @param storage        {@link HoodieStorage} for reading records.
    * @return {@link ClosableIterator<T>} that can return all records through iteration.
    */
-  public abstract ClosableIterator<T> getFileRecordIterator(
+  protected abstract ClosableIterator<T> getFileRecordIterator(
       StoragePath filePath, long start, long length, Schema dataSchema, Schema requiredSchema,
       HoodieStorage storage) throws IOException;
 
+  /**
+   * Gets the record iterator based on the type of engine-specific record representation from the
+   * file.
+   *
+   * @param file           {@link StorageFile} instance of a file.
+   * @param start          Starting byte to start reading.
+   * @param length         Bytes to read.
+   * @param dataSchema     Schema of records in the file in {@link Schema}.
+   * @param requiredSchema Schema containing required fields to read in {@link Schema} for projection.
+   * @param storage        {@link HoodieStorage} for reading records.
+   * @return {@link ClosableIterator<T>} that can return all records through iteration.
+   */
+  public final ClosableIterator<T> getFileRecordIterator(
+      StorageFile file, long start, long length, Schema dataSchema, Schema requiredSchema,
+      HoodieStorage storage) throws IOException {
+    if (file.getPathInfo() != null) {
+      return getFileRecordIterator(file.getPathInfo(), start, length, dataSchema, requiredSchema, storage);
+    } else {
+      return getFileRecordIterator(file.getStoragePath(), start, length, dataSchema, requiredSchema, storage);
+    }
+  }
+
+  /**
+   * Gets the record iterator based on the type of engine-specific record representation from the
+   * file.
+   *
+   * @param storagePathInfo {@link StoragePathInfo} instance of a file.
+   * @param start           Starting byte to start reading.
+   * @param length          Bytes to read.
+   * @param dataSchema      Schema of records in the file in {@link Schema}.
+   * @param requiredSchema  Schema containing required fields to read in {@link Schema} for projection.
+   * @param storage         {@link HoodieStorage} for reading records.
+   * @return {@link ClosableIterator<T>} that can return all records through iteration.
+   */
+  protected ClosableIterator<T> getFileRecordIterator(
+      StoragePathInfo storagePathInfo, long start, long length, Schema dataSchema, Schema requiredSchema,
+      HoodieStorage storage) throws IOException {
+    return getFileRecordIterator(storagePathInfo.getPath(), start, length, dataSchema, requiredSchema, storage);
```

Comment:
Adding to the above, only the path is used within the `StorageFile` instance.  So this util does not add much value.

Reply:
Since this is overridden by the `HiveHoodieReaderContext` and has a default implementation, I think it's OK to keep it now.